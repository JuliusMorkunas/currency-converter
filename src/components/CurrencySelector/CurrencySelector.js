import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import s from './CurrencySelector.module.scss';
import { useOnClickOutside } from '../../utils/hooks';
import Label from '../Label';

const CurrencySelector = ({
  currency = '',
  currencies = [],
  label = '',
  isOpen = false,
  onToggle,
  onSelect,
}) => {
  const ref = useRef(null);
  useOnClickOutside(ref, () => {
    isOpen && onToggle(false);
  });

  return (
    <div className={[s.Container, isOpen ? s.IsOpen : ''].join(' ')} ref={ref}>
      <Label onClick={() => onToggle(!isOpen)}>{label}</Label>
      <div className={s.InputContainer} onClick={() => onToggle(!isOpen)}>
        <div className={s.SelectedCurrencyContainer}>
          <div className={[s.CurrencyIcon, s[currency]].join(' ')} />
          <div className={s.CurrencyName}>{currency}</div>
        </div>
        <span className={s.DropdownArrow} />
      </div>
      {isOpen && (
        <div className={s.DropdownContainer}>
          <div className={s.Dropdown}>
            {currencies
              .filter(item => item !== currency)
              .map(currency => (
                <div
                  key={currency}
                  className={s.Option}
                  role="button"
                  tabIndex="0"
                  onClick={() => onSelect(currency)}
                >
                  <div className={[s.CurrencyIcon, s[currency]].join(' ')} />
                  <div className={s.CurrencyName}>{currency}</div>
                </div>
              ))}
          </div>
        </div>
      )}
    </div>
  );
};

CurrencySelector.propTypes = {
  currency: PropTypes.string,
  currencies: PropTypes.array,
  label: PropTypes.string,
  isOpen: PropTypes.bool,
  onToggle: PropTypes.func,
  onSelect: PropTypes.func,
};

export default CurrencySelector;
