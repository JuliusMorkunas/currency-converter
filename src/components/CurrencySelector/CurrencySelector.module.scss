@import '../../variables';

.Container {
  width: 100%;
}

.IsOpen {
  .DropdownArrow {
    transform: rotate(180deg);
  }
}

.InputContainer {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding-top: 6px;
  padding-bottom: 8px;
  cursor: pointer;
  border-bottom: 1px solid $c-blueish-border;
  outline: none;
}

.SelectedCurrencyContainer {
  display: flex;
  align-items: center;
}

.CurrencyName {
  font-size: 24px;
  line-height: 1;
  text-transform: uppercase;
  color: $c-black;
}

.DropdownArrow {
  width: 10px;
  height: 7px;
  background: transparent url('./icon-arrow-down.svg') no-repeat center;
  background-size: contain;
}

.DropdownContainer {
  position: relative;
  display: block;

  @media (max-width: $b-small) {
    position: fixed;
    z-index: 999;
    top: 0;
    left: 0;
    overflow-y: scroll;
    width: 100%;
    height: 100%;
    padding: 25px;
    background: rgba(0, 0, 0, 0.5);
  }
}

.Dropdown {
  position: absolute;
  z-index: 1;
  overflow-y: scroll;
  width: 100%;
  height: 34vh;
  border: 1px solid rgba(205, 225, 231, 0.3);
  background-color: $c-white;
  box-shadow: 0 2px 2px rgba(0, 0, 0, 0.18);

  @media (max-width: $b-small) {
    position: relative;
    height: 100%;
  }
}

.Option {
  display: flex;
  align-items: center;
  width: 100%;
  height: 40px;
  padding: 8px;
  cursor: pointer;
  transition: background-color 0.1s ease-in-out;
  border-bottom: 1px solid rgba(205, 225, 231, 0.3);
  background-color: $c-white;

  &:hover {
    background-color: rgba(205, 225, 231, 0.3);
  }

  .CurrencyName {
    font-size: 14px;
    color: $c-blackish;
  }
}

.CurrencyIcon {
  margin-right: 8px;
}

.aud,
.bgn,
.cad,
.chf,
.dkk,
.gbp,
.hkd,
.huf,
.ils,
.inr,
.mxn,
.nok,
.php,
.pln,
.ron,
.rub,
.sek,
.try,
.uah,
.usd,
.zar,
.eur {
  display: inline-block;
  overflow: hidden;
  text-align: left;
  text-indent: -9999px;
  background: url('./currency-flags-sprite.png') no-repeat;
}

.eur {
  width: 31px;
  height: 22px;
  background-position: -1px -0px;
}
.aud {
  width: 30px;
  height: 22px;
  background-position: -1px -23px;
}
.bgn {
  width: 30px;
  height: 22px;
  background-position: -1px -46px;
}
.cad {
  width: 30px;
  height: 22px;
  background-position: -1px -69px;
}
.chf {
  width: 30px;
  height: 22px;
  background-position: -1px -92px;
}
.dkk {
  width: 30px;
  height: 22px;
  background-position: -1px -115px;
}
.gbp {
  width: 30px;
  height: 22px;
  background-position: -1px -138px;
}
.hkd {
  width: 30px;
  height: 22px;
  background-position: -1px -161px;
}
.huf {
  width: 30px;
  height: 22px;
  background-position: -1px -184px;
}
.ils {
  width: 30px;
  height: 22px;
  background-position: -1px -207px;
}
.inr {
  width: 30px;
  height: 22px;
  background-position: -1px -230px;
}
.mxn {
  width: 30px;
  height: 22px;
  background-position: -1px -253px;
}
.nok {
  width: 30px;
  height: 22px;
  background-position: -1px -276px;
}
.php {
  width: 30px;
  height: 22px;
  background-position: -1px -299px;
}
.pln {
  width: 30px;
  height: 22px;
  background-position: -1px -322px;
}
.ron {
  width: 30px;
  height: 22px;
  background-position: -1px -345px;
}
.rub {
  width: 30px;
  height: 22px;
  background-position: -1px -368px;
}
.sek {
  width: 30px;
  height: 22px;
  background-position: -1px -391px;
}
.try {
  width: 30px;
  height: 22px;
  background-position: -1px -414px;
}
.uah {
  width: 30px;
  height: 22px;
  background-position: -1px -437px;
}
.usd {
  width: 30px;
  height: 22px;
  background-position: -1px -460px;
}
.zar {
  width: 30px;
  height: 22px;
  background-position: -1px -483px;
}
