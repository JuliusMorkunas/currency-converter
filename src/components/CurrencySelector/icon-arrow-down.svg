<svg width="10" height="7" xmlns="http://www.w3.org/2000/svg">
  <path d="M1.18.52L5 4.36 8.82.52 10 1.7l-5 5-5-5z" fill="#4A4A4A" fill-rule="evenodd"/>
</svg>
