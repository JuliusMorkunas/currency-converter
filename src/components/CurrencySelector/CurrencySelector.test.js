import React from 'react';
import { render } from '@testing-library/react';
import CurrencySelector from './CurrencySelector';

it('CurrencySelector should match snapshot', () => {
  const { asFragment } = render(<CurrencySelector />);
  expect(asFragment()).toMatchSnapshot();
});
