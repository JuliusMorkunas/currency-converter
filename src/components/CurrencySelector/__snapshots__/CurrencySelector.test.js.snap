// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencySelector should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Container "
  >
    <label
      class="Label"
    >
      :
    </label>
    <div
      class="InputContainer"
    >
      <div
        class="SelectedCurrencyContainer"
      >
        <div
          class="CurrencyIcon "
        />
        <div
          class="CurrencyName"
        />
      </div>
      <span
        class="DropdownArrow"
      />
    </div>
  </div>
</DocumentFragment>
`;
